dll/win32/shlwapi/utils.cpp

```diff
@@ -2,17 +2,35 @@
  * PROJECT:     ReactOS Shell
  * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
  * PURPOSE:     Implement shell light-weight utility functions
- * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ * COPYRIGHT:   Copyright 2023-2024 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
  */
 
 #define _ATL_NO_EXCEPTIONS
+
+/*
+ * HACK! These functions are conflicting with <shobjidl.h> inline functions...
+ */
+#define IShellFolder_GetDisplayNameOf _disabled_IShellFolder_GetDisplayNameOf_
+#define IShellFolder_ParseDisplayName _disabled_IShellFolder_ParseDisplayName_
+#define IShellFolder_CompareIDs _disabled_IShellFolder_CompareIDs_
+
 #include "precomp.h"
 #include <shellapi.h>
 #include <shlwapi.h>
-#include <shlwapi_undoc.h>
 #include <shlobj_undoc.h>
 #include <shlguid_undoc.h>
 #include <atlstr.h>
+
+/*
+ * HACK!
+ */
+#undef IShellFolder_GetDisplayNameOf
+#undef IShellFolder_ParseDisplayName
+#undef IShellFolder_CompareIDs
+
+#define SHLWAPI_ISHELLFOLDER_HELPERS /* HACK! */
```

Comment:
> that don't otherwise use a full-blown IShellFolder_.... interface

No, it seems to be for the shell to use when interacting with 3rd-party IShellFolders that it does not fully trust.